src/bsoncxx/document/element.cpp

```diff
@@ -69,7 +73,8 @@ bsoncxx::type element::type() const {
 stdx::string_view element::key() const {
     if (_raw == nullptr) {
         throw bsoncxx::exception{error_code::k_unset_element,
-                                 "cannot return the key from an uninitialized element"};
+                                 "cannot return the key from an uninitialized element with key \"" +
```

Comment:
The phrasing here is kind of ridiculous but I think that changing the behavior of `key()` to no longer thrown an exception when a nonexistent value is fetched would be problematic. Maybe we can just revert this particular message to what it was before the PR.

Reply:
> to no longer throw an exception when a nonexistent value is fetched would be problematic

I am not sure if there is a practical use-case to rely on `key()` throwing an exception. I think there is a risk changing `key()` to no longer throw an exception could break user assumptions. Example:

```c++
// insertAuthenticated inserts `doc` if `doc` contains the "authenticated" field.
bool insertAuthenticated(mongocxx::collection coll, bsoncxx::document::view doc) {
    try {
        // Check for the presence of the "authenticated" key before inserting.
        doc["authenticated"].key();
        coll.insert_one(doc);
    } catch (const bsoncxx::exception& e) {
        return false;
    }
}
```